convention-develocity-shared/convention-develocity-maven-extension/build.gradle.kts

```
@@ -23,3 +23,9 @@ publishing {
         }
     }
 }
+
+tasks.jar.configure {
+    from(zipTree(configurations.compileClasspath.get().first { it.name.contains("develocity-maven-extension") })) {
+        include("META-INF/maven/extension.xml")
+    }
+}
```

`tasks.jar.configure` is redundant in Kotlin DSL. Just `tasks.jar` is enough.

We can also avoid dependency resolution during configuration time by using `map`. 

```suggestion
tasks.jar {
    from(zipTree(configurations.runtimeClasspath.map { it.files.first { jar -> jar.name.contains("develocity-maven-extension") } })) {
        include("META-INF/maven/extension.xml")
    }
}
```

---

Very nice, thanks for the hint @erichaagdev 👏
I was just considering encapsulating the dependency resolution in a provider block but your suggestion is way more elegant 🫶

FYI I can't just accept the suggestion as it would sign it off with my private GH account 🤷‍♂️ I'll submit a new commit